packages/typespec-client-generator-core/src/types.ts

```diff
@@ -531,6 +536,54 @@ export function getSdkEnum(context: SdkContext, type: Enum, operation?: Operatio
   return sdkType;
 }
 
+function getSdkUnionEnumValues(
+  context: SdkContext,
+  type: UnionEnum,
+  enumType: SdkEnumType
+): SdkEnumValueType[] {
+  const values: SdkEnumValueType[] = [];
+  for (const [name, member] of type.flattenedMembers.entries()) {
+    const docWrapper = getDocHelper(context, member.variant);
+    values.push({
+      kind: "enumvalue",
+      name: typeof name === "string" ? name : `${member.value}`,
+      description: docWrapper.description,
+      details: docWrapper.details,
+      value: member.value,
+      valueType: enumType.valueType,
+      enumType,
+      nullable: false,
+    });
+  }
+  return values;
+}
+
+function getSdkUnionEnum(context: SdkContext, type: UnionEnum, operation?: Operation) {
+  let sdkType = context.modelsMap?.get(type.union) as SdkEnumType | undefined;
+  if (!sdkType) {
+    if (!type.union.name) throw new Error("Your union must be named in order to generate an enum");
```

Comment:
```suggestion
    if (!type.union.name) throw new Error("Union must be named in order to generate an enum");
```

Reply:
Should we include a couple of values to inform the user which union we are referring to in the error? For e.g. `Union with values a, b, c.... must be named in order to generate an enum`.

Reply:
We can, also there's actually no way we'll enter this code if the union isn't named, bc we then generate it as a union of literals bc we only call this function if the union is named. I honestly might just use `!` to avoid confusion